Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -1 +1,57 @@
 #include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+#include "Spire/Spire/Utility.hpp"
+
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const OrderImbalance& imbalance) {
+  m_last_published_imbalance = imbalance;
+  insert_sorted(imbalance);
+  m_imbalance_published_signal(imbalance);
+}
+
+void LocalOrderImbalanceIndicatorModel::insert(
+    const OrderImbalance& imbalance) {
+  insert_sorted(imbalance);
+}
+
+QtPromise<std::vector<Nexus::OrderImbalance>>
+    LocalOrderImbalanceIndicatorModel::load(const TimeInterval& interval) {
+  auto first = std::lower_bound(m_imbalances.begin(), m_imbalances.end(),
+    interval.lower(), [] (const auto& imbalance, const auto& timestamp) {
+      return imbalance.m_timestamp < timestamp;
+    });
+  auto last = std::upper_bound(m_imbalances.begin(), m_imbalances.end(),
+    interval.upper(), [] (const auto& timestamp, const auto& imbalance) {
+      return imbalance.m_timestamp > timestamp;
+    });
+  return QtPromise(
```

Comment:
What if ```interval``` is an open interval?

Reply:
Then the implementation probably gets pretty ugly. Is it possible to create a ClosedInterval type?

Reply:
```interval``` can be open or closed or half open/closed.